nautobot_device_lifecycle_mgmt/migrations/0021_migrate_to_core_software_models.py

```diff
@@ -0,0 +1,264 @@
+
+    # Migrate CustomField content type
+    for cf in CustomField.objects.filter(content_types=old_ct):
+        cf.content_types.add(new_ct)
+
+    # Migrate CustomLink content type
+    CustomLink.objects.filter(content_type=old_ct).update(content_type=new_ct)
+
+    # Migrate ExportTemplate content type - skip git export templates
+    ExportTemplate.objects.filter(content_type=old_ct, owner_content_type=None).update(content_type=new_ct)
+
+    # Migrate JobButton content type
+    for job_button in JobButton.objects.filter(content_types=old_ct):
+        job_button.content_types.add(new_ct)
+
+    # Migrate JobHook content type
+    for job_hook in JobHook.objects.filter(content_types=old_ct):
+        job_hook.content_types.add(new_ct)
+
+    # Migrate Note content type
+    Note.objects.filter(assigned_object_type=old_ct).update(assigned_object_type=new_ct)
+
+    # Migrate ObjectChange content type
+    ObjectChange.objects.filter(changed_object_type=old_ct).update(changed_object_type=new_ct)
+
+    # Migrate ObjectPermission content type
+    for object_permission in ObjectPermission.objects.filter(object_types=old_ct):
+        object_permission.object_types.add(new_ct)
+
+    # Migrate Relationship content type
+    Relationship.objects.filter(source_type=old_ct).update(source_type=new_ct)
+    Relationship.objects.filter(destination_type=old_ct).update(destination_type=new_ct)
+
+    # Migration RelationshipAssociation content type
+    RelationshipAssociation.objects.filter(source_type=old_ct).update(source_type=new_ct)
+    RelationshipAssociation.objects.filter(destination_type=old_ct).update(destination_type=new_ct)
+
+    # Migrate Status content type
+    for status in Status.objects.filter(content_types=old_ct):
+        status.content_types.add(new_ct)
+
+    # Migrate Tag content type
+    for tag in Tag.objects.filter(content_types=old_ct):
+        tag.content_types.add(new_ct)
+
+    # Migrate TaggedItem content type
+    TaggedItem.objects.filter(content_type=old_ct).update(content_type=new_ct)
+
+    # DLM forms are using a custom tag field that doesn't enforce content type. Fix that here if necessary.
+    for tag_id in TaggedItem.objects.filter(content_type=new_ct).values_list("tag_id", flat=True).distinct():
+        tag = Tag.objects.get(id=tag_id)
+        if not tag.content_types.filter(id=new_ct.id).exists():
+            tag.content_types.add(new_ct)
+
+    # Migrate WebHook content type
+    for web_hook in WebHook.objects.filter(content_types=old_ct):
+        web_hook.content_types.add(new_ct)
+
+
+def _migrate_hashing_algorithm(value):
+    # Attempt to map the hashing algorithm to one of the valid choices for dcim.SoftwareImageFile
+    similarity = {}
+    for choice in dcim_choices.SoftwareImageFileHashingAlgorithmChoices.values():
+        ratio = SequenceMatcher(lambda x: x not in ascii_letters + digits, value.lower(), choice.lower()).ratio()
+        similarity.setdefault(ratio, []).append(choice)
```

Comment:
Maybe add some comments here for the benefit of those less familiar with `difflib`?